File: SDKs/amazonaws.com/devicefarm/2015-06-23/go/pkg/models/shared/instanceprofile.go
package shared

// InstanceProfile
// Represents the instance profile.
type InstanceProfile struct {
	Arn                           *string  `json:"arn,omitempty"`
	Description                   *string  `json:"description,omitempty"`
	ExcludeAppPackagesFromCleanup []string `json:"excludeAppPackagesFromCleanup,omitempty"`
	Name                          *string  `json:"name,omitempty"`
	PackageCleanup                *bool    `json:"packageCleanup,omitempty"`
	RebootAfterUse                *bool    `json:"rebootAfterUse,omitempty"`
}

File: SDKs/amazonaws.com/devicefarm/2015-06-23/typescript/src/sdk/models/shared/instanceprofile.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";


// InstanceProfile
/** 
 * Represents the instance profile.
**/
export class InstanceProfile extends SpeakeasyBase {
  @Metadata({ data: "json, name=arn" })
  arn?: string;

  @Metadata({ data: "json, name=description" })
  description?: string;

  @Metadata({ data: "json, name=excludeAppPackagesFromCleanup" })
  excludeAppPackagesFromCleanup?: string[];

  @Metadata({ data: "json, name=name" })
  name?: string;

  @Metadata({ data: "json, name=packageCleanup" })
  packageCleanup?: boolean;

  @Metadata({ data: "json, name=rebootAfterUse" })
  rebootAfterUse?: boolean;
}
